src/node/convolver.rs

```diff
@@ -0,0 +1,353 @@
+use super::{AudioNode, ChannelConfig, ChannelConfigOptions, ChannelInterpretation};
+use crate::buffer::AudioBuffer;
+use crate::context::{AudioContextRegistration, BaseAudioContext};
+use crate::render::{AudioParamValues, AudioProcessor, AudioRenderQuantum, RenderScope};
+
+use crossbeam_channel::{Receiver, Sender};
+
+use realfft::{num_complex::Complex, RealFftPlanner};
+
+/// Scale buffer by an equal-power normalization
+fn normalize_buffer(buffer: &mut AudioBuffer) {
+    let gain_calibration = 0.00125;
+    let gain_calibration_sample_rate = 44100.;
+    let min_power = 0.000125;
+
+    // Normalize by RMS power.
+    let number_of_channels = buffer.number_of_channels();
+    let length = buffer.length();
+
+    let mut power: f32 = buffer
+        .channels()
+        .iter()
+        .map(|c| c.as_slice().iter().map(|&s| s * s).sum::<f32>())
+        .sum();
+
+    power = (power / (number_of_channels * length) as f32).sqrt();
+
+    // Protect against accidental overload.
+    if !power.is_finite() || power.is_nan() || power < min_power {
+        power = min_power;
+    }
+
+    let mut scale = 1. / power;
+
+    // Calibrate to make perceived volume same as unprocessed.
+    scale *= gain_calibration;
+
+    // Scale depends on sample-rate.
+    scale *= gain_calibration_sample_rate / buffer.sample_rate();
+
+    // True-stereo compensation.
+    if number_of_channels == 4 {
+        scale *= 0.5;
+    }
+
+    buffer
+        .channels_mut()
+        .iter_mut()
+        .for_each(|c| c.as_mut_slice().iter_mut().for_each(|s| *s *= scale))
+}
+
+/// `ConvolverNode` options
+//dictionary ConvolverOptions : AudioNodeOptions {
+//  AudioBuffer? buffer;
+//  boolean disableNormalization = false;
+//};
+#[derive(Clone, Debug, Default)]
+pub struct ConvolverOptions {
+    /// The desired buffer for the ConvolverNode
+    pub buffer: Option<AudioBuffer>,
+    /// The opposite of the desired initial value for the normalize attribute
+    pub disable_normalization: bool,
+    /// AudioNode options
+    pub channel_config: ChannelConfigOptions,
+}
+
+/// Processing node which applies a linear convolution effect given an impulse response.
```

Comment:
Maybe add some disclaimer, "This node is still a proof of concept and won't be usable in most real-world use-cases"